Tasgrid/tasgridWrapper.cpp

```diff
@@ -544,6 +547,29 @@ bool TasgridWrapper::getInterWeights(){
 
     return true;
 }
+bool TasgridWrapper::getDiffWeights(){
+    auto x = readMatrix(xfilename);
+    if (x.getStride() != (size_t) grid.getNumDimensions()){
+        cerr << "ERROR: grid is set for " << grid.getNumDimensions() << " dimensions, but " << xfilename << " specifies " << x.getStride() << endl;
+        return false;
+    }
+    if (x.empty()){
+        cerr << "ERROR: no points specified in " << xfilename << endl;
+        return false;
+    }
+    int num_w = x.getNumStrips();
+    int num_p = grid.getNumPoints() * grid.getNumDimensions();
+    Data2D<double> result(num_p, num_w);
+
+    #pragma omp parallel for
+    for(int i=0; i<num_w; i++)
+        grid.getDifferentiationWeights(x.getStrip(i), result.getStrip(i));
+
+    writeMatrix(outfilename, num_w, (int) num_p, result.getStrip(0));
```

Comment:
* `num_p` is already `int`, no need to convert
* use `result.data()` it is more clear that you are dumping all data, as opposed to the first strip only

Reply:
Fixed.